docs/utxo-hd/utxo-db/utxo-db-api.tex

```diff
@@ -678,102 +684,101 @@ \subsection{Access to multiple (logical) ledger states}
-to keep open the option to use parallelism later then it is necessary for the
+We may not make use of parallelism in an initial implementation, but if we are
+to keep open the option to use parallelism later, then it is necessary for the
 interface between the application and the disk storage layer to expose the
-opportunities for parallelism. Thus we wish to find an interface that allows
+opportunities for parallelism. Thus, we wish to find an interface that allows
 for I/O parallelism.
 
 It is worth keeping in mind how much parallel I/O we need to saturate a modern
-SSD. It is on the order of 32 -- 64 concurrent I/O operations being performed
+SSD. It is in the order of 32 -- 64 concurrent I/O operations being performed
 at all times. A useful abstraction is to think of it as a queue of in-progress
-operation where new I/O operations are added at one end and results arrive
+operations where new I/O operations are added at one end and results arrive
 eventually at the other end, and the queue should be kept sufficiently full to
-saturate the SSD's throughput. Due to the high throughput and timing jitter it
-is better to `over-fill' the queue by some amount, e.g. $2\times$. That is, in
-order to ensure the SSD queue `depth' does not drain to below 64, at the
-application level it may be necessary to aim to keep double that number of
-operations in progress. The appropriate amount to use can be tuned based on I/O
+saturate the SSD's throughput. Due to the high throughput and timing jitter, it
+is better to `over-fill' the queue by some amount, e.g., $2\times$. That is, in
+order to ensure the SSD queue `depth' does not drain to below 64, it may be necessary to aim to keep double that number of
+operations in progress at the application level. The appropriate amount to use can be tuned based on I/O
 profiling tools, but the overall point is clear: to fully exploit the
-throughput of an SSD we need to keep a substantial number of operations in
+throughput of an SSD, we need to keep a substantial number of operations in
 progress at once -- and on a continuous basis.
 
-Since blockchains are mostly linear in nature (being a chain) the opportunities
+Since blockchains are mostly linear in nature (being a chain), the opportunities
 for I/O parallelism come from batching and pipelining.
 \begin{description}
-\item[Batching:] This is submitting a batch of I/O operations and waiting and
-                collecting them all. For example we could submit all the I/O
+\item[Batching:] This is submitting a batch of I/O operations and waiting to
+                collect them all. For example, we could submit all the I/O
                 reads for a single block in one go.
 
                 A block with 64 transactions with 2 UTxO inputs each would
-                generate 128 read IOPS. So we can see that large enough blocks
+                generate 128 read I/O operations. So, we can see that large enough blocks
```

Comment:
Just to double check: should the *per-seconds* part be removed here?

Reply:
Yes, we're only talking about the number of IO operations in a single block produces, which isn't related to time